dev-env/lib/dade-dump-profile

```diff
@@ -99,7 +107,12 @@ source "$(dirname "${BASH_SOURCE[0]}")/ensure-nix"
 echo "source $(dirname "${BASH_SOURCE[0]}")/ensure-nix"
 
 # Expose our tools in /dev-env/bin and our version of nixpkgs
-echo "export NIX_USER_CONF_FILES=\"${DADE_DEVENV_DIR}/etc/nix.conf:\${NIX_USER_CONF_FILES:-}:\${XDG_CONFIG_HOME:-\${HOME}/.config}/nix/nix.conf\""
+if [[ "$("${DADE_DEVENV_DIR}/bin/semver" compare "$(get_nix_version)" '2.4.0')" -ge 0 ]]; then
```

Comment:
maybe the loop comes from the fact that we use semver from dev-env but this is dev-env setup? So you dev-env while you dev-env and you have a bad time?

Reply:
Oh, damn, you are probably right. Any idea how we can fix this?

Reply:
I can reproduce it by opening a new shell. Will fix.

Reply:
You could either do a more naive version comparison yourself (similar to the old version check we had for too old nix versions), call into nix directly, something like `$(nix-build nix -A tools.semver --no-out-link)/bin/semver` or go for the full revert after all. I’m fine with all 3.

Reply:
I ended up going for option 2 before I saw your message. Let me know if you think it's reasonable.

Reply:
looks good :+1: